crates/wac-types/src/targets.rs

```diff
@@ -0,0 +1,122 @@
+use crate::{ExternKind, ItemKind, SubtypeChecker, Types, WorldId};
+use std::collections::{BTreeMap, BTreeSet};
+use std::fmt;
+
+/// The result of validating a component against a target world.
+pub type TargetValidationResult = Result<(), TargetValidationReport>;
+
+/// The report of validating a component against a target world.
+#[derive(Debug, Default)]
+pub struct TargetValidationReport {
+    /// Imports present in the component but not in the target world.
+    pub imports_not_in_target: BTreeSet<String>,
```

Comment:
For API stability sake would it make sense to make these fields private and add accessors that just return `impl Iterator<Item = &str>`?